lib/rec.c

```diff
@@ -657,6 +686,14 @@ tlog_rec(struct tlog_errs **perrs, uid_t euid, gid_t egid,
     num = json_object_get_int64(obj);
     latency = (unsigned int)num;
 
+    /* Read the rate limit*/
+    if (!json_object_object_get_ex(conf, "rate", &obj)) {
+        tlog_errs_pushs(perrs, "Rate is not specified");
+        grc = TLOG_RC_FAILURE;
+        goto cleanup;
+    }
+    check_rate = json_object_get_int64(obj);
+
```

Comment:
Since you're passing `conf` to `tlog_rec_create_log_sink` anyway, there is no need to read the rate here and pass it as an extra argument, just read it in `tlog_rec_create_log_sink`.